test/jdk/jdk/jfr/api/consumer/TestManyRecordings.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.jfr.api.consumer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import jdk.jfr.Event;
+import jdk.jfr.Recording;
+import jdk.jfr.consumer.RecordedClass;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedThread;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.jfr.Events;
+import jdk.test.lib.jfr.TestClassLoader;
+
+/**
+ * This is not a unit test, but a testapp that can be
+ * used to provoke bugs or performance issues when many classes or
+ * class loaders are being used.
+ *
+ * Example usage:
+ *
+ *  java -XX:StartFlightRecording:settings=profile
+ *       TestManyClasses <class-loader-count> <classes-per-class-loader>>
+ */
+public class TestManyRecordings {
+
+    private List<ClassLoader> loaders = new ArrayList<>();
+
+    static class TestEvent {
+        Class<> theClass;
+    }
+    public static void main(String[] args) throws Throwable {
+        int classLoaderCount = Integer.parseInt(args[0]);
+        int classCount = Integer.parseInt(args[1]);
+        for (int i = 0; i <classCount; i++) {
```

Comment:
Did you mean classLoaderCount here instead of classCount? Also, please make sure there is a space between < and classLoaderCount.

Reply:
The JFR "tests" looks strange. I would expect at a test called TestManyRecording to start recordings, not created a lot of classes, similar to TestManyClasses. How is this related to Loom?  Could this be a merge gone bad?

Reply:
It seems that test/jdk/jdk/jfr/api/consumer/TestManyClasses.java, TestManyRecordings.java, and TestParse.java were added for another JFR event (nothing to do with VirtualThreadPinned). @egahlin has contributed some cleanup and these files are removed.